File: Cpp/1342_Number_of_Steps_to_Reduce_a_Number_to_Zero.cpp
class Solution {
public:
    int numberOfSteps(int num) {
        int step = 0;
        while(num != 0){
            step++;
            num = num%2 == 0 ? num/2 : num-1;
            /*
            if(num%2 == 0){
                num /= 2;
            }
            else{
                num -= 1;
            }
            */
        }
        return step;
    }
};

File: CS/1342_Number_of_Steps_to_Reduce_a_Number_to_Zero.cs
public class Solution {
    public int NumberOfSteps(int num) {
        int result = 0;
        while(num != 0){
            if(num%2 == 0){
                num /= 2;
            }
            else{
                num -= 1;
            }
            result++;
        }
        return result;
    }
}
